clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/evaluation/EvaluationManager.kt

```diff
@@ -0,0 +1,503 @@
+
+            // Sort eligible client-side in-app notifications by priority.
+            sortByPriority(eligibleInApps).forEach { inApp ->
+                // Check if the in-app notification should not be suppressed.
+                if (!shouldSuppress(inApp)) {
+                    // Save suppressed client-side in-app IDs before returning if there were updates.
+                    if (updated) {
+                        saveSuppressedClientSideInAppIds()
+                    } // save before returning
+
+                    // Update the Time to Live (TTL) for the in-app notification.
+                    updateTTL(inApp)
+                    // Return a JSONArray containing the current in-app notification.
+                    return JSONArray().also { it.put(inApp) }
+                } else {
+                    // Update the flag, suppress the in-app, and continue processing.
+                    updated = true
+                    suppress(inApp)
+                }
+            }
+            // Save suppressed client-side in-app IDs before returning if there were updates.
+            if (updated) {
+                saveSuppressedClientSideInAppIds()
+            } // save before returning
+
+            // Return an empty JSONArray if no eligible in-app notifications are displayed.
+        }.run { return JSONArray() }
+    }
+
+    /**
+     * Evaluates a list of in-app notifications against the provided event, considering triggers and limits.
+     *
+     * This method iterates through the given list of in-app notifications and checks for matches with the event's triggers.
+     * If a match is found, the method increments the associated triggers, checks against limits, and processes accordingly.
+     * Eligible in-app notifications are added to a list for further consideration.
+     *
+     * @param event The [EventAdapter] representing the event triggering the in-app notification evaluation.
+     * @param inappNotifs A list of JSONObjects representing in-app notifications to be evaluated.
+     * @param clearResource A function to clear resources, invoked when a limit indicates data should be discarded.
+     *                     Accepts a URL parameter
+     *
+     * @return A list of JSONObjects representing in-app notifications that meet the criteria for display.
+     */
+    @VisibleForTesting
+    internal fun evaluate(
+        event: EventAdapter,
+        inappNotifs: List<JSONObject>,
+        clearResource: (url: String) -> Unit = {}
+    ): List<JSONObject> {
+        val eligibleInApps: MutableList<JSONObject> = mutableListOf()
+
+        for (inApp in inappNotifs) {
+            val campaignId = inApp.optString(Constants.INAPP_ID_IN_PAYLOAD)
+
+            val matchesTrigger =
+                triggersMatcher.matchEvent(getWhenTriggers(inApp), event)
+            if (matchesTrigger) {
+                Logger.v("INAPP", "Triggers matched for event ${event.eventName} against inApp $campaignId")
```

Comment:
ideally "INAPP" should be a contant in inapps package and used across the feature, lets do it as tech debt

Reply:
tech debt